.gitlab-ci.yml

```diff
@@ -0,0 +1,71 @@
+image: 'node:10.14'
+
+stages:
+  - setup
+  - test
+  - build
+  - release
+
+# Fallback to support `includes`
+cache:
+  key: $CI_COMMIT_REF_SLUG
+  paths:
+    - node_modules/
+    - lib/
+
+.shared:
+  artifacts:
+    paths:
+      - lib/
+      - package.json
+      - CHANGELOG.md
```

Comment:
artifacts here are files that can be downloaded by the upcoming jobs in a pipeline.

This is not superclean here since the `include`s don't use the `extends` syntax. The set of artificats is preliminary but I'm sure that we don't need a `README` as artifact right now.

Reply:
Ok, I know what the general purpose of artifacts is (to share data between pipelines). What are we actually _using_ artifacts here for? What’s the reason a subsequent pipeline needs access to a previous pipeline’s output in this repo?

Reply:
ok you sounded like you need a primer on artifacts.

You're completely right, these lines are not final. As you know, the general purpose is to not do redundant computations but rather cache outputs via `cache` and `artifact`. Before we now excerpt the `.gitlab-ci.yml` line by line, i'd encourage us to be happy about a green pipeline and make it nice in another MR. I will also figure out a good `extend` structure for our internal pipelines in this sprint.

You'll hear back regarding this in another MR

Reply:
Created https://github.com/strong-config/node/issues/13

Reply:
alright, let's discuss in more detail in the ticket.